mathesar/views/api.py

```diff
@@ -126,6 +126,36 @@ def type_suggestions(self, request, pk=None):
         col_types = get_table_column_types(table)
         return Response(col_types)
 
+    @action(methods=['post'], detail=True)
+    def previews(self, request, pk=None):
+        table = self.get_object()
+        serializer = TablePreviewSerializer(data=request.data)
+        if not serializer.is_valid():
+            return Response(serializer.errors)
```

Comment:
You're not returning an error status code here.

Reply:
This could just be `serializer.is_valid(raise_exception=True)`?

Reply:
Thanks, good to know.

Reply:
@mathemancer You're still not returning a status code here. Actually, you don't need the if statement at all, you can just have `serializer.is_valid(raise_exception=True)` be the statement and then you don't need to return a `Response`.